#!/usr/bin/env python

PACKAGE = "wheeltec_arm_pick"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x_p", double_t, 0, "X PID PARAM", 0.35, -2.0, 2.0)

gen.add("x_i", double_t, 0, "X PID PARAM", 0.0, -1.0, 1.0)

gen.add("x_d", double_t, 0, "X PID PARAM", 0.02, -1.0, 1.0)

gen.add("y_p", double_t, 0, "Y PID PARAM", 0.35, -2.0, 2.0)

gen.add("y_i", double_t, 0, "Y PID PARAM", 0.0, -1.0, 1.0)

gen.add("y_d", double_t, 0, "Y PID PARAM", 0.02, -1.0, 1.0)

gen.add("color_location_x", double_t, 0, "COLOR LOCATION", -0.13, -2.00, 2.00)

gen.add("color_location_y", double_t, 0, "COLOR LOCATION", 0.00, -2.00, 2.00)

gen.add("arm_upper_and_lower", double_t, 0, "arm state ", 0.00, 0.00, 2.00)
 
gen.add("hand_open_and_close", double_t, 0, "hand state ", 0.00, 0.00, 2.00)

gen.add("car_start", double_t, 0, "car state ", 0.00, 0.00, 1.00)

# Second arg is node name it will run in (doc purposes only), third is generated filename prefix
exit(gen.generate(PACKAGE, "wheeltec_arm_pick", "params"))

